### PR TITLE
feat: add script to generate the theme

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Add script to generate the theme:
- create the `tools` folder in which we will have the helper scripts/tools for component, like theme/styles generator
- script creates the two css files, with reset css `{name}.reset.css` and without reset css `{name}.css` - https://github.com/asyncapi/asyncapi-react/issues/368
- script takes two parameters, first is a path to output css file like `./output/index.css` and the second parameter (optional) is the path to the custom configuration for TailwindCSS. We deep merge the default configuration with custom configuration.
- add `theming.md` documentation. 

We can use the script like 

```sh
node ./library/tools/styles/script.js ./output/index.css
// or
node ./library/tools/styles/script.js ./output/index.css ./custom.config.js
```

In the future we can move this script to our cli with `generate theme` commands like:

```bash
asyncapi generate theme --config=... --output=...
```

Is it a good idea? Feel free to comment!

Problems I found:
- dependencies... at the moment we can only add all needed dependencies to generate the styles like `tailwindcss`, `postcss` and their plugins to `dependencies` array, then people will able to generate the styles using script in their project like: `node node_modules/@asyncapi/react-component/tools/styles/script.js ...`, of course we will end up with bigger package, but the component itself in application won't change the weight, because these dependencies are a pure devDependencies that aren't used in component directly, but only in `tools/styles` folder.

**Related issue(s)**
Resolves https://github.com/asyncapi/asyncapi-react/issues/370
Resolves https://github.com/asyncapi/asyncapi-react/issues/368

### CODE REVIEW

File: docs/features/theming.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+where:
+- `<output-file>` is a relative path for the output CSS file. It must contains the name of the CSS file. Example: `./output/custom-theme.css`
+- `<config-file-path>` is a relative path to the custom [TailwindCSS configuration file](https://tailwindcss.com/docs/configuration).
+
+Important notes:
+
+- The custom configuration is deeply merged with the [default configuration file](../../library/tools/styles/tailwind.config.js).
+
+- To generate the production-ready, tree-skahable, CSS styles you should run the script with `NODE_ENV=production` environment:
```
Comment (fmvilas) [inline docs/features/theming.md:29]:
```suggestion
- To generate the production-ready, tree-shakable, CSS styles you should run the script with `NODE_ENV=production` environment:
```

File: docs/features/theming.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  NODE_ENV=production node ./tools/styles/script.js ./output/custom-theme.css ./custom-config.js
+  ```
+
+  Generating production styles will add a `.min.css` suffix to the generated files.
+
+- Script will generate two separate CSS files, one with `.reset.css` suffix with [Preflight](https://tailwindcss.com/docs/preflight) styles onboard and second without `Preflight` onboard. If you use your own base styles in your project, you should use this one with the `.reset.css` suffix.
+
+- If you installed the `@asyncapi/react-component` package in your project, script should be run with the `./node_modules/@asyncapi/react-component/tools/styles/script.js` path in a root of project.
+
+## Change manually styles
```
Comment (fmvilas) [inline docs/features/theming.md:41]:
```suggestion
## Changing the styles manually
```

File: docs/features/theming.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+> **NOTE:** This is not the recommended method - better to use the [configuration file](#generate-theme-from-configuration-file).
+
+To manually change the style you can write your own CSS properties for appropriate TailwindCSS class. For example, to change the a very frequently used color, you can override it by CSS class:
```
Comment (fmvilas) [inline docs/features/theming.md:45]:
```suggestion
To manually change the style you can write your own CSS properties for the appropriate TailwindCSS class. For example, to change the very frequently used color, you can override it by CSS class:
```

File: docs/features/theming.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+```css
+.text-gray-700 {
+  color: red;
+}
+```
+
+> **NOTE:** Remember that the given override must be loaded after loading the styles from the component.
```
Comment (fmvilas) [inline docs/features/theming.md:53]:
```suggestion
> **NOTE:** Remember that your styles must be loaded after the default styles of the component.
```

File: library/tools/styles/script.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
+const postcss = require('postcss');
```
Comment (fmvilas) [inline library/tools/styles/script.js:1]: Can we come up with a better name for this file? `script` seems too broad and doesn't say anything. Maybe call it `theme-generator` or `css-generator` or `styles-generator` or ...

Comment (magicmatatjahu) [reply library/tools/styles/script.js]: You're right, I changed to `theme-generator`, thanks!

**Comment by jonaslagoni** [comment]
@magicmatatjahu you have a small conflict 😄 

@fmvilas are you gonna finish reviewing this PR?

**Comment by magicmatatjahu** [comment]
@jonaslagoni Thanks for info! Fixed.

**Review by fmvilas — APPROVED** [review]
Looks good to me 👍

**Comment by derberg** [comment]
@magicmatatjahu pinging in case you forgot, although I'm pretty sure you didn't 😄

**Pull request closed without merging**